crates/trie/sparse/src/parallel_trie.rs

```diff
@@ -47,3 +355,40 @@ pub struct SparseSubtrie {
     /// The map from paths to sparse trie nodes within this subtrie.
     nodes: HashMap<Nibbles, SparseNode>,
 }
+
+impl SparseSubtrie {
+    fn new(path: Nibbles) -> Self {
+        Self { path, nodes: Default::default() }
+    }
+
+    /// Internal implementation of the method of the same name from ParallelSparseTrie. This method
+    /// may return leaf paths/values which need to be added to the ParallelSparseTrie's values
+    /// HashMap.
+    fn reveal_node_or_hash(
+        &mut self,
+        path: Nibbles,
+        node_or_hash: TrieNodeOrHashToReveal,
+        masks: TrieMasks,
+    ) -> SparseTrieResult<Option<Vec<(Nibbles, Vec<u8>)>>> {
+        match reveal_node_or_hash(&mut self.nodes, path, node_or_hash, masks)? {
+            RevealNodeOutcome::Continue => Ok(None),
+            RevealNodeOutcome::LeavesRevealed(leaves) => Ok(Some(leaves)),
+            RevealNodeOutcome::RevealChildNodes(children) => {
```

Comment:
`RevealNodeOutcome::RevealChildNodes` may have hashes for non-leaf nodes, and we will return them as leaves, I think this is incorrect